rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoaderImpl.java

```diff
@@ -230,6 +232,20 @@ private void setupPrecompiledContractsStorage(Repository repository) {
         }
     }
 
+    public static InputStream loadGenesisFile(String fileName) {
+        InputStream inputStream = GenesisLoaderImpl.class.getResourceAsStream("/genesis/" + fileName);
+        if (inputStream != null) {
+            return inputStream;
+        }
+
+        try {
+            return Files.newInputStream(Paths.get(fileName));
+        } catch (Exception e) {
+            logger.error("Could not load file due to:", e);
+            return null;
+        }
```

Comment:
Since you are opening creating and opening the stream here, I would advise to close it in a `finally` clause, just in case an exception occurs.

Reply:
Erm... The stream is not open yet, so the exception can only happen before or whilst opening it. Therefore, there is nothing to close, is it?

Reply:
True, but if you pass the return from this `Files.newInputStream(Paths.get(fileName));` to the variable `inputStream` you will have. 

It's just on the case that some error happened during the opening of this stream,  It's probably very unlikely to happen and probably the GC would handle, but it was a suggestion just to play safe.

But you know what, this is really too much. And looking now, I noticed that the stream will be closed when we are reading the json if something weird happens. But now it occurred to me that you could just catch the Exception strictly necessary, Exception is too broad and we want to catch what is the real root cause and why is not covered if something happens. 

```
        try {
            return Files.newInputStream(Paths.get(fileName));
        } catch (IOException | RuntimeException e) {
            logger.error("Cannot read genesis json file");

            throw new RuntimeException("Cannot open genesis block configuration file", e);
        }```

Reply:
outdated